src/amazonica/aws/kinesis.clj

```diff
@@ -267,8 +278,15 @@
         {:keys [processor deserializer checkpoint credentials dynamodb-adaptor-client?]
          :or   {checkpoint 60
                 deserializer unwrap}} opts
+        checkpoint-strategy (cond
+                              (number? checkpoint) :timeout
+                              ;; not sure of a good name for this strategy:
+                              (false? checkpoint) :boolean
```

Comment:
Are the `:boolean` case handled in the processor function above? It's hard to read if the behavior of the `:timeout` case here is the same as the current behavior.